admin/src/linter.rs

```diff
@@ -134,4 +134,29 @@ impl Linter {
 
         Ok(())
     }
+
+    /// Checks if a crate with this name is present on crates.io
+    fn name_exists_on_crates_io(&self, name: &str) -> Result<bool, Error> {
+        // This contains a homebrew crates.io API client.
+        // It was created because `crates_io_api` is bloated (async!)
+        // and breaks any time crates.io changes any fields at all,
+        // even the ones we don't use. And we literally need ONE field.
+
+        #[derive(serde::Deserialize)]
+        struct CrateResponse {
+            #[serde(alias = "crate")] // "crate" is a reserved keyword and cannot be used as a name
+            crate_info: CrateInfo, // there are more fields, but this is the only one we need
+        }
+
+        #[derive(serde::Deserialize)]
+        struct CrateInfo {
+            name: String, // there are more fields, but this is the only one we need
+        }
+
+        let url = format!("https://crates.io/api/v1/crates/{}", name);
+        let response: CrateResponse = self.http_client.get(&url).call()?.into_json()?;
+        // FIXME: I've ported this equality check from legacy code
+        // and I have no idea what it does or why it's needed
```

Comment:
It has to do with name normalization. A request for https://crates.io/api/v1/crates/serde-json will return `serde_json` data an we want to catch use a non-canonical name.